src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/FilesTab.java

```diff
@@ -250,17 +253,17 @@ public static boolean equal(Object a, Object b) {
 		return a == b || (a != null && a.equals(b));
 	}
 	  
-	public void getTargetBundleAndDisplay(String entityId, Long versionNumber) {
+	public void getTargetBundleAndDisplay(String entityId, final Long versionNumber) {
 		//only ask for it if we are showing a different entity/version
-		if (equal(currentEntityId,entityId) && equal(currentVersionNumber, versionNumber)) {
+		if (equal(currentEntityId,entityId) && equal(shownVersionNumber, versionNumber)) {
 			return;
 		}
 		
 		currentEntityId = entityId;
-		currentVersionNumber = versionNumber;
+		shownVersionNumber = versionNumber;
 		synAlert.clear();
 		int mask = ENTITY | ANNOTATIONS | PERMISSIONS | ENTITY_PATH | HAS_CHILDREN | ACCESS_REQUIREMENTS | UNMET_ACCESS_REQUIREMENTS | FILE_HANDLES | ROOT_WIKI_ID | DOI | FILE_NAME;
-		AsyncCallback<EntityBundle> callback = new AsyncCallback<EntityBundle>() {
+		final AsyncCallback<EntityBundle> callback = new AsyncCallback<EntityBundle>() {
```

Comment:
Two callbacks are unnecessary, there is really just one async call.  Please rewrite so that it is a single callback (change callback to take EntityBundlePlus).